src/util/generateOpenGraphImage.js

```diff
@@ -0,0 +1,153 @@
+            imagePaint.setImageFilter(CanvasKit.ImageFilter.MakeMatrixTransform(CanvasKit.Matrix.scaled(xRatio, yRatio), { filter: CanvasKit.FilterMode.Linear }, null));
+            const imageLeft = isRtl
+                ? (1 / xRatio) * (width - margin['inline-start']) - logoW
+                : (1 / xRatio) * margin['inline-start'];
+            canvas.drawImage(img, imageLeft, (1 / yRatio) * margin['block-start'], imagePaint);
+        }
+    }
+    // Load and configure font families.
+    const fontMgr = await fontManager.get(fonts);
+		let Styles = {
+			textAlign: isRtl ? CanvasKit.TextAlign.Right : CanvasKit.TextAlign.Left,
+			textStyle: textStyle(font.title),
+			textDirection: isRtl ? CanvasKit.TextDirection.RTL : CanvasKit.TextDirection.LTR,
+	}
+    if (fontMgr) {
+        // Create paragraph with initial styles and add title.
+        const paragraphStyle = new CanvasKit.ParagraphStyle(Styles);
+        const paragraphBuilder = CanvasKit.ParagraphBuilder.Make(paragraphStyle, fontMgr);
+        paragraphBuilder.addText(decodeHTMLStrict(title));
+        // Add small empty line betwen title & description.
+        paragraphBuilder.pushStyle(new CanvasKit.TextStyle({ fontSize: padding / 3, heightMultiplier: 1 }));
+        paragraphBuilder.addText('\n\n');
+        // Add description.
+        paragraphBuilder.pushStyle(new CanvasKit.TextStyle(textStyle(font.description)));
+        paragraphBuilder.addText(decodeHTMLStrict(description));
+        // Draw paragraph to canvas.
+        const para = paragraphBuilder.build();
+        const paraWidth = width - margin['inline-start'] - margin['inline-end'] - padding;
+        para.layout(paraWidth);
+        const paraLeft = isRtl
+            ? width - margin['inline-start'] - para.getMaxWidth()
+            : margin['inline-start'];
+        const minTop = margin['block-start'] + logoHeight + (logoHeight ? padding : 0);
+        const maxTop = minTop + (logoHeight ? padding : 0);
+        const naturalTop = height - margin['block-end'] - para.getHeight();
+        const paraTop = Math.max(minTop, Math.min(maxTop, naturalTop));
+				
+        canvas.drawParagraph(para, paraLeft, paraTop);
+				if(code){
+					const codeStyle = new CanvasKit.ParagraphStyle(Styles)
+					const codeText = CanvasKit.ParagraphBuilder.Make(codeStyle, fontMgr)
+					codeText.addText(decodeHTMLStrict(code))
+					codeText.pushStyle(new CanvasKit.TextStyle(textStyle(font.description)));
+					const codeTop = height - minTop + padding
+					const buildCodeBlock = codeText.build();
+					buildCodeBlock.layout(paraWidth)
+					console.log(buildCodeBlock.getHeight())
+					const rectInputs = [paraLeft,codeTop,paraWidth, codeTop + buildCodeBlock.getHeight() + 13,50,50,50,50,50,50,50,50]
+					const paintRect = new CanvasKit.Paint()
+					paintRect.setColor(bgGradient.map(x=>x/2))
+					canvas.drawRRect(rectInputs,paintRect)
+					canvas.drawParagraph(buildCodeBlock,paraLeft + padding / 2, codeTop + 5)
```

Comment:
This is the additions that I had made to apply the code block  to the canvas

Reply:
The changes where just to see if it was possible and how it would look, Im happy that this can help add some flexibility to the `astro-og-image` after spending time with it I kinda like it, albeit still an awful library, Ive already began an upstream branch on the repo with some of the changes, and making some of these api changes